From 1cf170260b78e550ed0a0cd3b729527852de0991 Mon Sep 17 00:00:00 2001
From: Ronan Pigott <ronan@rjp.ie>
Date: Fri, 18 Jul 2025 15:12:31 +0200
Subject: [PATCH] manager: s/deserialized_subscribed/subscribed_as_strv

Now that this field may get populated at runtime, the deserialized name
is misleading. Change the name to reflect its updated purpose.

(cherry picked from commit e1315a621ae26473fcc9cd0d6013836f5f498d40)

Related: RHEL-75081
---
 src/core/dbus.c    | 6 +++---
 src/core/manager.c | 8 ++++----
 src/core/manager.h | 2 +-
 3 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/src/core/dbus.c b/src/core/dbus.c
index 1b8bb44eda..9f439dcb23 100644
--- a/src/core/dbus.c
+++ b/src/core/dbus.c
@@ -908,8 +908,8 @@ int bus_init_api(Manager *m) {
         if (r < 0)
                 return log_error_errno(r, "Failed to set up API bus: %m");
 
-        (void) bus_track_coldplug(bus, &m->subscribed, /* recursive= */ false, m->deserialized_subscribed);
-        m->deserialized_subscribed = strv_free(m->deserialized_subscribed);
+        (void) bus_track_coldplug(bus, &m->subscribed, /* recursive= */ false, m->subscribed_as_strv);
+        m->subscribed_as_strv = strv_free(m->subscribed_as_strv);
         m->api_bus = TAKE_PTR(bus);
 
         r = manager_enqueue_sync_bus_names(m);
@@ -1079,7 +1079,7 @@ static void destroy_bus(Manager *m, sd_bus **bus) {
                 r = bus_track_to_strv(m->subscribed, &subscribed);
                 if (r < 0)
                         log_warning_errno(r, "Failed to serialize api subscribers, ignoring: %m");
-                strv_free_and_replace(m->deserialized_subscribed, subscribed);
+                strv_free_and_replace(m->subscribed_as_strv, subscribed);
 
                 m->subscribed = sd_bus_track_unref(m->subscribed);
         }
diff --git a/src/core/manager.c b/src/core/manager.c
index c5d906e42e..37bdac8b1d 100644
--- a/src/core/manager.c
+++ b/src/core/manager.c
@@ -1349,7 +1349,7 @@ Manager* manager_free(Manager *m) {
         free(m->switch_root_init);
 
         sd_bus_track_unref(m->subscribed);
-        strv_free(m->deserialized_subscribed);
+        strv_free(m->subscribed_as_strv);
 
         rlimit_free_all(m->rlimit);
 
@@ -1659,8 +1659,8 @@ int manager_startup(Manager *m, FILE *serialization, FDSet *fds) {
         manager_setup_bus(m);
 
         /* Now that we are connected to all possible busses, let's deserialize who is tracking us. */
-        (void) bus_track_coldplug(m->api_bus, &m->subscribed, false, m->deserialized_subscribed);
-        m->deserialized_subscribed = strv_free(m->deserialized_subscribed);
+        (void) bus_track_coldplug(m->api_bus, &m->subscribed, false, m->subscribed_as_strv);
+        m->subscribed_as_strv = strv_free(m->subscribed_as_strv);
 
         /* Third, fire things up! */
         manager_coldplug(m);
@@ -3382,7 +3382,7 @@ int manager_deserialize(Manager *m, FILE *f, FDSet *fds) {
                         exec_runtime_deserialize_one(m, val, fds);
                 else if ((val = startswith(l, "subscribed="))) {
 
-                        if (strv_extend(&m->deserialized_subscribed, val) < 0)
+                        if (strv_extend(&m->subscribed_as_strv, val) < 0)
                                 log_oom();
                 } else {
                         ManagerTimestamp q;
diff --git a/src/core/manager.h b/src/core/manager.h
index 98d381bc5b..e713250238 100644
--- a/src/core/manager.h
+++ b/src/core/manager.h
@@ -215,7 +215,7 @@ struct Manager {
         considered subscribes, since they last for very short only,
         and it is much simpler that way. */
         sd_bus_track *subscribed;
-        char **deserialized_subscribed;
+        char **subscribed_as_strv;
 
         /* This is used during reloading: before the reload we queue
          * the reply message here, and afterwards we send it */
